include/libvfio-user.h

```diff
@@ -268,33 +246,106 @@ typedef enum {
 } vfu_pci_type_t;
 
 /**
- * Setup PCI configuration space header data. This function must be called only
- * once per libvfio-user context.
+ * Initialize the context for a PCI device. Returns an initialized buffer for a
+ * type 0 PCI configuration space of the needed size.
+ *
+ * This buffer consists of the standard 64-byte PCI header and the remaining
+ * space depending on @pci_type, either PCI_CFG_SPACE_SIZE or
+ * PCI_CFG_SPACE_EXP_SIZE.
  *
  * @vfu_ctx: the libvfio-user context
- * @id: Device and vendor ID
- * @ss: Subsystem vendor and device ID
- * @cc: Class code
  * @pci_type: PCI type (convention PCI, PCI-X mode 1, PCI-X mode2, PCI-Express)
  * @revision: PCI/PCI-X/PCIe revision
+ */
+vfu_pci_hdr_t *
+vfu_pci_init(vfu_ctx_t *vfu_ctx, vfu_pci_type_t pci_type, int revision UNUSED);
```

Comment:
We should add the header type as a parameter and state that only type 0 is currently supported. This way if someone implements other header types in the future we won't have to change the API.

Reply:
I'm not convinced it will ever make sense to have the other types, but sure.